Core/Types/CkanModule.cs

```diff
@@ -366,6 +376,33 @@ private void DeSerialisationFixes(StreamingContext like_i_could_care)
             name = name ?? string.Empty;
         }
 
+        static CkanModule()
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            using (var stream = assembly.GetManifestResourceStream(SchemaPath))
+            using (var reader = new StreamReader(stream))
+            {
+                var result = reader.ReadToEnd();
+                MetadataSchema = JsonSchema4.FromSampleJson(result);
+            }
+        }
+
+        private static readonly string SchemaPath = "CKAN.CKAN.schema";
+        private static readonly JsonSchema4 MetadataSchema;
+
+        private static async Task<IList<string>> ValidateAgainstSchema(string json)
+        {
+            JsonSchema4 schema = await JsonSchema4.FromJsonAsync(json);
+
+            IList<string> errorList = new List<string>();
+            var errors = schema.Validate(MetadataSchema.ToJson());
```

Comment:
If I'm reading this correctly, we process the schema from a JSON string to a JsonSchema4 object in the static constructor, and then back from object to a JSON string in the validation function.

Can we store the return value of the ToJson function, so we don't have to pay the cost of that conversion for every CkanModule object? Presumably we can ensure that the CKAN.schema file bundled in the assembly is valid separately whenever we modify it.